### ISSUE TITLE
Xbox One S Controller, Fix for Select not working via Bluetooth 

### ISSUE THREAD

**Issue by Throway123** [RetroPie/RetroPie-Setup#2023]
Hello, I think everybody at this point know that if you want to use Xbox One S controller with the lastest firmware 3.1.1221.0 via Bluetooth you first need to disable ERTM (Enhanced Re-Transmission Mode a L2CAP Bluetooth stack feature) because Bluez/Bluez-firmware/pi-bluetooth (who knows) dont support this feature and always show some type of error when you try to connect,, you can find the error messages all over the internet.

But even after you disable ERTM and connect successful you cant fully use the controller, why? Because now the controller is detect as a HID Android Device (because of the lastest firmware change) changing the mapping and some other stuff, basically, now Select, xbox button send out of range commands.

Raspberian and RetroPie installs SDL 2.0.5+5rpi package to handle low level access to audio, keyboard, mouse, joystick, this is the lastest version avaliable to install but it dont work with Xbox One S Controller, why? Because it was released 8 months ago, and the lastest Xbox One S commit on SDL project was 5 months ago (http://hg.libsdl.org/SDL/log?rev=xbox), which probably fix everything related to xbox one controller, since it says "The XBox One S controller sends keys outside the standard joystick button range ".

I tested it on my Rasp 3, removed SDL 2.0.5, installed SDL from the source, and now Select button on Xbox One S controller WORKS, BUT, in libreretro emulators via Retroarch, probably because of how old XPAD is, it takes a few minutes to the controller start working in general (like 2min without any key working), last time I fixed it installing again XPAD from source in retropie setup, but now because of time and misconfiguration of my part I didnt tested, I dont trully know the relashionship between XPAD and SDL so I dont know more about it.

So TL;DR version of all this, it is possible to Retropie/Raspbian team to install by default SDL from source, or to update SDL binary to source version with xbox one s fixes, and maybe take a look at XPAD or Bluez? If not, can someone please teach me how to remove SDL binary and install SDL from the source without everytime some other package which uses SDL try to ''overwrite'' my manual installation because SDL package isnt installed officially.

Sorry for the long text but Im trying to fix/workaround this problem for a whole week.

Thanks!

**Comment by joolswills** [comment]
Known issue and there is a PR here - https://github.com/RetroPie/SDL-mirror/pull/4 however, this isn't a RetroPie bug but down to the last SDL2 release not having this patch. I need to re-do our SDL patchsets, so reduce the number of branches we have, but wanted to wait until the next official SDL2 release, rather than maintaining yet another backported patch.
